ddlite.py

```diff
@@ -1132,39 +1135,63 @@ def plot_learning_diagnostics(self, mu_opt, f1_opt):
   def set_use_lfs(self, use=True):
     self.use_lfs = bool(use)
 
-  def get_log_odds(self, subset=None):
-    """
-    Get the array of predicted link function values (continuous) given weights
-    Return either all candidates, a specified subset, or only validation/test set
-    """
-    start = 0 if self.use_lfs else self.num_lfs()
-    if self.X is None or self.w is None:
+  def _log_odds(self, X, w, start, subset):
+    if X is None or w is None:
       raise ValueError("Inference has not been run yet")
     if subset is None:
-      return self.X[:, start:].dot(self.w[start:])
+      return X[:, start:].dot(w[start:])
     if subset is 'test':
-      return self.X[self.test(), start:].dot(self.w[start:])
+      return X[self.test(), start:].dot(w[start:])
     if subset is 'validation':
-      return self.X[self.validation(), start:].dot(self.w[start:])
+      return X[self.validation(), start:].dot(w[start:])
     try:
-      return self.X[subset, start:].dot(self.w[start:])
+      return X[subset, start:].dot(w[start:])
     except:
       raise ValueError("subset must be 'test', 'validation', or an array of\
                        indices 0 <= i < {}".format(self.num_candidates()))
 
+  def get_log_odds(self, subset=None):
+    """
+    Get the array of predicted link function values (continuous) given weights
+    Return either all candidates, a specified subset, or only validation/test set
+    """
+    start = 0 if self.use_lfs else self.num_lfs()
+    return self._log_odds(self.X, self.w, start, subset)
+    
+  def get_log_odds_lf(self, subset=None):
+    """
+    Get the array of predicted link function values (continuous) given LF weights
+    Return either all candidates, a specified subset, or only validation/test set
+    """
+    return self._log_odds(self.lf_matrix.tocsr(), self._lf_w, 0, subset)
+
```

Comment:
I'm confused about what these mean- `get_log_odds` and `get_log_odds_lf`... are these functions returning the marginal probabilities of the candidates?  If so the naming here is a bit off.  Also I don't know what "link function" means here?